### PR TITLE
Improve DX

### PR DESCRIPTION
**Pull request by jaredjj3**
This PR improves the developer experience of this project. The main feature is the dev server:

image

- Adds a dev server that makes it easy to see rendered Vexflow.
- Adds the ability to write tests using `jest`.
- Adds VSCode settings to lint and format on save.
- Adds a `git` pre-commit hook to lint, format, and check that the project is buildable.
- Updates the structure of the project.
- Documents on how to develop the project in the README.

### CODE REVIEW

**Comment by jaredjj3** [comment]
@rvilarl, after this gets merged I recommend that you delete and clone the repo again to avoid artifacts of the old setup from being accidentally committed.

File: dev/public/examples/beam.xml [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
-      <supports element="accidental" type="yes"/>
-      <supports element="beam" type="yes"/>
-      <supports element="print" attribute="new-page" type="yes" value="yes"/>
-      <supports element="print" attribute="new-system" type="yes" value="yes"/>
-      <supports element="stem" type="yes"/>
+      <supports element="accidental" type="yes" />
+      <supports element="beam" type="yes" />
+      <supports element="print" attribute="new-page" type="yes" value="yes" />
+      <supports element="print" attribute="new-system" type="yes" value="yes" />
+      <supports element="stem" type="yes" />
```
Comment (rvilarl) [inline dev/public/examples/beam.xml:13]: for these two examples it is fine, but probably we should avoid that eslint / prettier looks into this directory

File: package.json [hunk 2]
```diff
@@ -1,18 +1,66 @@
...
+  "files": [
+    "dist/**/*"
+  ],
+  "keywords": [
+    "vexflow",
+    "musicxml"
+  ],
+  "license": "MIT",
   "dependencies": {
     "vexflow": "^4.0.1-rc.0"
```
Comment (rvilarl) [inline package.json:17]: 4.0.1 is already out

File: tsconfig.json [hunk 3]
```diff
@@ -1,16 +1,14 @@
...
-    "forceConsistentCasingInFileNames": true /* Disallow inconsistently-cased references to the same file. */,
-    "moduleResolution": "node",
-    "lib": ["dom", "es2017"],
-    "removeComments": false
-
-  }
-}
\ No newline at end of file
+    "target": "es5",
+    "module": "commonjs",
```
Comment (rvilarl) [inline tsconfig.json:4]: Why do you prefer commojs to esmodules?

**Review by rvilarl — APPROVED** [review]

File: dev/public/examples/beam.xml (hunk 1 above)
Comment (jaredjj3) [reply dev/public/examples/beam.xml:13]: Done. `.eslintrc.json` has `"ignorePatterns": ["*.xml"]` and I added a `.prettierignore` entry for `*.xml`.

File: package.json (hunk 2 above)
Comment (jaredjj3) [reply package.json:17]: Updated.

File: tsconfig.json (hunk 3 above)
Comment (jaredjj3) [reply tsconfig.json:4]: I don't have a preference, that was the value from the config I copied. I changed it to `esnext`.

**Pull request merged**